### ISSUE TITLE
Fix false positives for comments within selector lists in selector-type-case

### ISSUE THREAD

**Issue by XhmikosR** [stylelint/stylelint#3698]
> Which rule, if any, is the bug related to?

` selector-type-case`

> What CSS is needed to reproduce the bug?

```less
.alert-dismissable, // The misspelled .alert-dismissable was deprecated in 3.2.0.
.alert-dismissible {
  padding-right: (@alert-padding + 20);

  // Adjust close link position
  .close {
    position: relative;
    top: -2px;
    right: -21px;
    color: inherit;
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

https://github.com/twbs/bootstrap/blob/97b95a524e7a413dca4278ee1cc9570c38ff5f25/grunt/.stylelintrc

> Which version of stylelint are you using?

`9.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Grunt

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Less

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

```
41:24 × Expected "The" to be "the" selector-type-case
```

v9.5.0 seems to be fine. After updating to 9.6.0 I'm getting this.

**Comment by hudochenkov** [comment]
I think it's fixed in less parser [3.0.2](https://github.com/shellscape/postcss-less/releases/tag/v3.0.2) and PostCSS [7.0.3+](https://github.com/postcss/postcss/releases/tag/7.0.3). Could you try reinstalling stylelint? If you use `package-lock.json` you might need `npm uninstall stylelint` first.

**Comment by XhmikosR** [comment]
@hudochenkov: stylelint 9.6.0 does not use postcss-less 3.x. It's using 2.0.0.

**Comment by jeddy3** [comment]
@XhmikosR Thanks for the report. This looks to be a regression. There's a pending PR, https://github.com/stylelint/stylelint/pull/3687, to update to a major new release of the Less parser. So you might not have access to `postcss-less-parser@3` yet.

As a workaround, I suspect moving the comment out of the selector list and placing it before the rule-set will remove the violation. Alternatively, you can temporarily turn off the `selector-type-case` rule around this snippet of code using the `/* stylelint-disable selector-type-case */` command.

**Comment by XhmikosR** [comment]
Yeah, already worked around it but I thought I'd report this :)

**Comment by jeddy3** [comment]
Closing as stylistic and relating to custom syntax.
